src/sentry/tsdb/redis.py

```diff
@@ -37,6 +37,32 @@
 )
 
 
+class SuppressionWrapper(object):
+    """\
+    Wraps a context manager and prevents exceptions raised by exiting the
+    wrapped context manager from propagating. Exceptions that were raised in
+    the context, however, are still propagated.
+    """
+
+    def __init__(self, wrapped, types=(Exception,)):
+        self.wrapped = wrapped
+        self.types = types
+
+    def __enter__(self):
+        return self.wrapped.__enter__()
+
+    def __exit__(self, *args):
+        if any(arg is not None for arg in args):
+            # propagate exception raised in managed context
+            return
+
+        try:
+            return self.wrapped.__exit__(*args)
+        except self.types:
+            # suppress exception raised when exiting wrapped manager
```

Comment:
I'd personally like to log this but I feel like that'd result in a compounding problem…?